route_linux.go

```diff
@@ -102,22 +157,28 @@ func routeHandle(route *Route, req *nl.NetlinkRequest, msg *nl.RtMsg) error {
 // Equivalent to: `ip route show`.
 // The list can be filtered by link and ip family.
 func RouteList(link Link, family int) ([]Route, error) {
+	var rf *RouteFilter
+	if link != nil {
+		rf = &RouteFilter{
+			Oif:      link.Attrs().Index,
+			FlagMask: RT_FILTER_OIF,
+		}
+	}
+	return RouteListFiltered(family, rf)
+}
+
+// RouteListFiltered gets a list of routes in the system filtered with specified rules.
+// All rules must be defined in RouteFilter struct
+func RouteListFiltered(family int, filter *RouteFilter) ([]Route, error) {
```

Comment:
Rather than having a separate route filter object that duplicates all the fields, I think it makes more sense to have two parameters,  RouteListFiltered(family int, filter *Route, filterMask uint64)